src/Ether.Api/Types/ADPasswordValidator.cs

```diff
@@ -30,19 +30,18 @@ public Task ValidateAsync(ResourceOwnerPasswordValidationContext context)
             }
 
             var domain = domainAndName[0];
-            var userName = context.UserName.Substring(context.UserName.IndexOf('\\'));
+            var userName = domainAndName[1];
             if (type == ContextType.Machine)
             {
                 domain = Environment.MachineName;
             }
 
             using (var principal = new PrincipalContext(type, domain))
             {
-                var isValid = principal.ValidateCredentials(userName, context.Password);
+                var isValid = principal.ValidateCredentials(userName, context.Password, ContextOptions.Negotiate);
                 if (isValid)
                 {
-                    var userNameToSearch = type == ContextType.Machine ? Environment.UserName : context.UserName;
```

Comment:
User can login as not current windows user, as another local user. In this case we cannot use Environment.UserName